extensions/amp-base-carousel/1.0/carousel-context.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {createContext} from '../../../src/preact';
+
+const CarouselContext = createContext(
+  /** @type {BaseCarouselDef.ContextProps} */ ({})
```

Comment:
Hmm. But now we do need to provide `setSlideCount` and `setCurrentSlide` here - right? They'll just be empty functions, but then (a) we don't need to test for them everywhere, and (b) it self-documents.

Reply:
Happy to put `setSlideCount`, however `setCurrentSlide` should fallback to the one given by `useState` if the `CarouselContext` is not used. One option could be to use `useState` in `carousel-context` as the default value instead, WDYT?

Reply:
I see. I didn't think about that. Ok. In that case, i'm all good with the undefined function fields and checks. Sorry for the confusion.

Reply:
We can still define `setSlideCount`, just not `setCurrentSlide`! I did this in [ae93511](https://github.com/ampproject/amphtml/pull/29674/commits/ae9351197acd8bc5bf0da3baa19da341de2eaf0f).

Reply:
Sounds good. Let's go with that.